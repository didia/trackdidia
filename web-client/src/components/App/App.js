import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import AppBar from 'material-ui/AppBar'
import NavLink from '../shared/NavLink'

let defaultStyles = {
    navLink: {
        textDecoration: 'none',
        color: 'inherit'
    }
}

class App extends Component {
    render() {
        return (
            <MuiThemeProvider>
                <div className="trackdidia">
                    <AppBar title={<NavLink to="/" onlyActiveOnIndex style={defaultStyles.navLink}>Life Dashboard</NavLink>} className="app-bar" />
                    <div className="page-content">
                        {this.props.children}
                    </div>
                </div>
            </MuiThemeProvider>
        );
    }
}

export default App;
