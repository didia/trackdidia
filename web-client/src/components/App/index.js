/**
 * Created by didia on 16-10-15.
 */
import App from './App';

export default App;
