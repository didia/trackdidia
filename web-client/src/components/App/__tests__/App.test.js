import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('has not unexpectedly changed', () => {
  const component = renderer.create(
      <App />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
