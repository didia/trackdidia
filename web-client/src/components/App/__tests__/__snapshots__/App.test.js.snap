exports[`test has not unexpectedly changed 1`] = `
<div
  className="trackdidia">
  <div
    className="app-bar"
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#00bcd4",
        "borderRadius": "0px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "flex",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "paddingLeft": 24,
        "paddingRight": 24,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
        "zIndex": 1100
      }
    }>
    <button
      disabled={false}
      href={undefined}
      onBlur={[Function anonymous]}
      onClick={[Function anonymous]}
      onFocus={[Function anonymous]}
      onKeyDown={[Function anonymous]}
      onKeyUp={[Function anonymous]}
      onMouseDown={[Function onMouseDown]}
      onMouseEnter={[Function anonymous]}
      onMouseLeave={[Function anonymous]}
      onMouseOut={[Function anonymous]}
      onMouseUp={[Function onMouseUp]}
      onTouchEnd={[Function onTouchEnd]}
      onTouchStart={[Function onTouchStart]}
      onTouchTap={[Function anonymous]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "marginLeft": -16,
          "marginRight": 8,
          "marginTop": 8,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48
        }
      }
      tabIndex={0}
      type="button">
      <div
        onMouseDown={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        onMouseUp={[Function anonymous]}
        onTouchEnd={[Function anonymous]}
        onTouchStart={[Function anonymous]}>
        <svg
          onMouseEnter={[Function anonymous]}
          onMouseLeave={[Function anonymous]}
          style={
            Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fill": "#ffffff",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24
            }
          }
          viewBox="0 0 24 24">
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
        </svg>
      </div>
    </button>
    <div
      onTouchTap={[Function anonymous]}
      style={
        Object {
          "boxFlex": 1,
          "color": "#ffffff",
          "flex": "1",
          "fontSize": 24,
          "fontWeight": 400,
          "height": 64,
          "letterSpacing": 0,
          "lineHeight": "64px",
          "margin": 0,
          "muiPrepared": true,
          "overflow": "hidden",
          "paddingTop": 0,
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap"
        }
      }>
      <a
        onClick={[Function bound handleClick]}
        style={
          Object {
            "color": "inherit",
            "textDecoration": "none"
          }
        }>
        Life Dashboard
      </a>
    </div>
  </div>
  <div
    className="page-content" />
</div>
`;
